src/util/arith_tools.cpp

```diff
@@ -182,13 +180,8 @@ constant_exprt from_integer(
     ieee_float.from_integer(int_value);
     return ieee_float.to_expr();
   }
-
-  {
+  else
```

Comment:
I think this case should be `UNREACHABLE` rather than `PRECONDITION(false)`.

Reply:
I'd stick to it -- this is really something imposed on the caller, and not something internally guaranteed.

Reply:
Ok!